### PR TITLE
Gridbehaviour

### PR DESCRIPTION
**Pull request by DylanGuidry95**

### CODE REVIEW

File: Assets/Scripts/EditorNodeUpdate.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EditorNodeUpdate : MonoBehaviour
+{
+    [ContextMenu("UpdateNodes")]
```
Comment (chuwilliamson) [inline Assets/Scripts/EditorNodeUpdate.cs]: please use buttons

File: Assets/Scripts/NodeBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class NodeBehaviour : MonoBehaviour
+{
+    public enum NodeType
+    {
+        TurrentBase, Wall, Path
```
Comment (chuwilliamson) [inline Assets/Scripts/NodeBehaviour.cs]: Turrent? check spelling

File: Assets/Scripts/NodeBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            return;
+        this.GetComponent<Renderer>().material.color = (isHovering) ? HighlightColor : DefaultColor;
+    }
+
+    public void SelectNode()
+    {
+        if(_NodeType != NodeType.TurrentBase)
+            return;
+        IsSelected = !IsSelected;
+        this.GetComponent<Renderer>().material.color = (IsSelected) ? SelectedColor : DefaultColor;
```
Comment (chuwilliamson) [inline Assets/Scripts/NodeBehaviour.cs]: do not use "this" keyword

**Comment by DylanGuidry95** [comment]
all feedback has been implemented

**Review by chuwilliamson — APPROVED** [review]

**Pull request merged**